### PR TITLE
mysql: how to migrate mysql database using the console

### PR DESCRIPTION
**Pull request by wojcik-dorota**
https://aiven.atlassian.net/browse/DOC-177
This is the guide on how to use the console migration tool to migrate MySQL databases.

Based on https://www.figma.com/file/CLVBPTf0A9DPeXrigqAThO/Database-migration?node-id=369%3A62896&t=nqpc1mFG7YBVJtUa-0

### CODE REVIEW

File: docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+   If you start a new migration using the same connection details when your target database is not empty, the migration tool truncates your target database and an existing data set gets overwritten with the new data set.
+
+Related reading
+---------------
+
+- :doc:`Migrate to Aiven for MySQL from an external MySQL </docs/products/mysql/howto/migrate-from-external-mysql>`
+- :doc:`About aiven-db-migrate </docs/products/postgresql/concepts/aiven-db-migrate>`
+- :doc:`Migrate to Aiven for PostgreSQL® with aiven-db-migrate </docs/products/postgresql/howto/migrate-aiven-db-migrate>`
+- :doc:`Migrate to Aiven for PostgreSQL® with pg_dump and pg_restore </docs/products/postgresql/howto/migrate-pg-dump-restore>`
+- :doc:`Migrate between PostgreSQL® instances using aiven-db-migrate in Python </docs/products/postgresql/howto/run-aiven-db-migrate-python>`
```
Comment (packi) [inline docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst:247]: If this is intentional I think those links are ok, but I was wondering why we have links to PostgreSQL here

File: docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+.. _start-over-mysql:
+
+Start over
+----------
+
+If you :ref:`stop a migration process <stop-migration-mysql>`, you cannot restart the same process. Still, the data already migrated is retained in the target database.
+
+.. warning::
+   
+   If you start a new migration using the same connection details when your target database is not empty, the migration tool truncates your target database and an existing data set gets overwritten with the new data set.
```
Comment (packi) [inline docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst:238]: Maybe put an emphasis on **target**?

File: docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+  * Existing on-premise MySQL databases
+  * Cloud-hosted MySQL databases
+  * Managed MySQL database clusters on Aiven.
+
+* Console migration tool can use either logical replication or physical replication.
+
+  * Logical replication allows the continuous migration, which is recommended and used by default in the tool, hence, also detailed in this guide. With this method, data transfer is possible not only for the data that has already been there in the source database when triggering the migration but also for any data written to the source database during the migration.
+
+  * Physical replication allows the other migration method supported in the migration tool, ``mysqldump``. With this method, current contents of the database are copied into a file and transferred to the target database. Any changes written to the source database during the migration are not transferred. When you trigger the migration setup in the console and initial checks detect that your source database does not support the logical replication, you are notified about it via wizard. To continue with the migration, you can select the alternative ``mysqldump`` migration method in the wizard.
```
Comment (packi) [inline docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst:23]: I wouldn't call this one physical replication it's a dump/restore

Comment (packi) [reply docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst:23]: Farther down it's referred to as **one-time snapshot (dump method)**

**Review by packi — CHANGES_REQUESTED** [review]
We should probably address the **physical replication** comment, the rest are stylistic and just things I noticed. Caveat: I didn't try the whole procedure myself.

File: docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst (hunk 3 above)
Comment (wojcik-dorota) [reply docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst:23]: > 

In those two points I compare two types of replication, logical vs physical, not migration is secondary here. And dump method is the migration method, not replication type. So we need a type of replication that is supported for the dump method. Any idea? If it's not physical replication, what replication is supported for the dump method?
Also, physical/logical replication types I borrowed from DO article on the same feature (https://docs.digitalocean.com/products/databases/mysql/how-to/migrate/#logical-replication-vs-physical-replication)

Comment (packi) [reply docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst:23]: Replication in itself is continuously streaming data. Physical or logical only defines if the changes are on a block- or on a statement-level. `mysqldump` doesn't replicate at all, it exports (dumps) the database into a text-file and re-imports it (restore).

In the DO article they do have it a bit wrong as well, as replication always means that the database is being streamed, one way or another.

Comment (wojcik-dorota) [reply docs/products/mysql/howto/migrate-db-to-aiven-via-console.rst:23]: @packi , many thanks for this clarification :-) Updating accordingly

**Review by divdby0 — APPROVED** [review]

**Review by packi — APPROVED** [review]

**Pull request merged**